src/shared/port-provider.js

```diff
@@ -71,23 +71,23 @@ export class PortProvider {
     this._emitter = new TinyEmitter();
 
     /**
-     * Map of source frame to channels that have been created for this frame.
+     * IDs of port requests that have been handled.
      *
      * This is used to avoid responding to the same request multiple times.
      * Guest frames in particular may send duplicate requests (see comments in
      * PortFinder).
      *
-     * @type {WeakMap<Window, Set<Channel>>}
+     * @type {Set<string>}
      */
-    this._sentChannels = new WeakMap();
+    this._handledRequests = new Set();
```

Comment:
There is no expiry of used request IDs here. We just rely on the total number being small enough that it doesn't use significant memory to remember them forever.